rclpy/package.xml

```diff
@@ -14,9 +14,13 @@
   <depend>rmw_implementation</depend>
   <depend>rcl</depend>
 
+  <exec_depend>ament_index_python</exec_depend>
+
   <test_depend>ament_cmake_nose</test_depend>
+  <test_depend>ament_index_python</test_depend>
```

Comment:
Actually I think it is a problem with topological ordering. Basically without the `test_depend` then `launch` isn't built before `rclpy`'s tests are run. This wouldn't be important except that topological order seems to influence what goes into a package's local_setup.sh file, which get sourced from the `cmake__test.[sh|bat]` files. The dependencies get calculated here using topological order:

https://github.com/ament/ament_tools/blob/188a30869e599fb1b8c5f74073952c97bef7d640/ament_tools/verbs/build/cli.py#L240-L253

The `context.build_dependencies` is assigned here:

https://github.com/ament/ament_tools/blob/188a30869e599fb1b8c5f74073952c97bef7d640/ament_tools/verbs/build_pkg/cli.py#L354-L355

And then they are used here to generate the `cmake__test.*` files:

https://github.com/ament/ament_tools/blob/188a30869e599fb1b8c5f74073952c97bef7d640/ament_tools/build_types/ament_python.py#L264-L269
and
https://github.com/ament/ament_tools/blob/188a30869e599fb1b8c5f74073952c97bef7d640/ament_tools/build_types/ament_python.py#L291-L299

The topological ordering considers `test_depend`'s as part of the build dependencies here:

https://github.com/ament/ament_tools/blob/188a30869e599fb1b8c5f74073952c97bef7d640/ament_tools/topological_order.py#L48-L50

So, with only the `exec_depend` it doesn't work because of the resulting topological order. I don't think this is the "right" behavior.

Sure enough, `launch` is no where in list of things to source for rclpy's test file on failing job (whereas on my local machine it is in the list):

From: http://ci.ros2.org/job/ci_linux/ws/ws/build/rclpy/cmake__test.sh/*view*/
Full text (in case the workspace goes): https://gist.github.com/wjwwood/8f749aedd155e62da87e8664ae6d2438